aten/src/ATen/ExpandUtils.cpp

```diff
@@ -149,9 +151,10 @@ std::vector<int64_t> infer_dense_strides(IntArrayRef tensor_sizes, IntArrayRef t
   // all dimensions with 0 stride won't move. This is the same behavior as TensorIterator.
   // eg. Given tensor with size/stride (6, 5, 4, 3, 2)/(6, 0, 120, 0, 1), the initial `perm`
   //     is (4, 3, 2, 1, 0) and the sorted `perm` will be (4, 3, 0, 1, 2)
-  for (int i = 1; i < ndim; ++i) {
-    int dim1 = i;
-    for (int dim0 = i - 1; dim0 >= 0; --dim0) {
+  for (const auto i : c10::irange(1, ndim)) {
+    auto dim1 = i;
+    for (const auto j : c10::irange(i)) {
+      auto dim0 = i - 1 - j;
```

Comment:
Wouldn't that achieve the same results but will be at least one arithmetic op faster?
```suggestion
    for (const auto j : c10::irange(1, i + 1)) {
      auto dim0 = i - j;
```

Reply:
I'd be wary about this line of thinking: modern compilers do so many optimization passes that a micro-optimization such as this is unlikely to have any detectable impact on the final performance of the code. I think the most important question is: which way best communicates intent?

Reply:
Thanks, they shows similar readability to me but the one suggested by @malfet didn't rely on compiler optimization and I will follow this suggestion.

Reply:
@frank-dong-ms : The `c10::irange` abstraction only works because the compiler performs an astounding number of optimizations to the point where speculating about which bounds are better is probably an exercise in futility :-)

Reply:
@r-barnes I disagree with you here. I.e. although modern compiler can probably detect that `std::sqrt(x*x)` is equivalent to `std::abs(x)` I would prefer to have a later in the code.
And similarly,  `for(int x=1; x < 11; ++x) { std::cout << x  << std::endl; }` is preferable to `for(int x=0; x < 10; x++) { std::cout << x + 1 << std::endl; }`

Reply:
@frank-dong-ms thank you for applying the suggestion.